adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -467,6 +467,11 @@ func (b *BM25Searcher) createTerm(N float64, filterDocIds helpers.AllowList, que
 		termResult.exhausted = true
 		return termResult, docMapPairsIndices, nil
 	}
+
+	if len(allMsAndProps) > 1 {
+		sort.Sort(ById(docMapPairs))
+	}
```

Comment:
You'd also need to apply the new order to docMapPairsIndices if `additionalExplanations`is set - otherwise the explanations (freq etc) are coming from the wrong entry. (Could you also add a test? :) )